routers/repo/migrate.go

```diff
@@ -144,35 +174,28 @@ func MigratePost(ctx *context.Context) {
 		err = migrations.IsMigrateURLAllowed(remoteAddr, ctx.User)
 	}
 	if err != nil {
-		if models.IsErrInvalidCloneAddr(err) {
-			ctx.Data["Err_CloneAddr"] = true
-			addrErr := err.(*models.ErrInvalidCloneAddr)
-			switch {
-			case addrErr.IsProtocolInvalid:
-				ctx.RenderWithErr(ctx.Tr("repo.mirror_address_protocol_invalid"), tpl, &form)
-			case addrErr.IsURLError:
-				ctx.RenderWithErr(ctx.Tr("form.url_error"), tpl, &form)
-			case addrErr.IsPermissionDenied:
-				if addrErr.LocalPath {
-					ctx.RenderWithErr(ctx.Tr("repo.migrate.permission_denied"), tpl, &form)
-				} else if len(addrErr.PrivateNet) == 0 {
-					ctx.RenderWithErr(ctx.Tr("repo.migrate.permission_denied_blocked"), tpl, &form)
-				} else {
-					ctx.RenderWithErr(ctx.Tr("repo.migrate.permission_denied_private_ip"), tpl, &form)
-				}
-			case addrErr.IsInvalidPath:
-				ctx.RenderWithErr(ctx.Tr("repo.migrate.invalid_local_path"), tpl, &form)
-			default:
-				log.Error("Error whilst updating url: %v", err)
-				ctx.RenderWithErr(ctx.Tr("form.url_error"), tpl, &form)
-			}
-		} else {
-			log.Error("Error whilst updating url: %v", err)
-			ctx.RenderWithErr(ctx.Tr("form.url_error"), tpl, &form)
-		}
+		ctx.Data["Err_CloneAddr"] = true
+		handleMigrateRemoteAddrError(ctx, err, tpl, form)
 		return
 	}
 
+	form.LFS = form.LFS && setting.LFS.StartServer
+
+	if form.LFS && len(form.LFSEndpoint) > 0 {
+		ep := lfs.DetermineEndpoint("", form.LFSEndpoint)
+		if ep == nil {
+			ctx.Data["Err_LFSEndpoint"] = true
+			ctx.RenderWithErr(ctx.Tr("repo.migrate.invalid_lfs_endpoint"), tpl, &form)
+			return
+		}
+		err = migrations.IsMigrateURLAllowed(form.LFSEndpoint, ctx.User)
```

Comment:
I think this really should be:

```suggestion
		err = migrations.IsMigrateURLAllowed(ep.String(), ctx.User)
```